starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,65 @@
 // Names and Input
 
+console.log("I'm Ready!");
+
+var hacker1 = "Celine";
+console.log("The driver's name is " + hacker1);
+
+var hacker2 = "Christina";
+console.log("The navigator's name is " + hacker2);
 
 //Conditionals
 
+if (hacker1.length < hacker2.length){
+  console.log("Yo, navigator got the longest name , it has " + hacker2.length + " characters");
+}
+else if (hacker1.length > hacker2.length){
+  console.log("the Driver has the longest name, it has " + hacker1.length + " characters");
+} else {
+  console.log("wow, you both got equally long names," + hacker1.length + " characters");  
+}
+
+var toUpperCaseString = "";
+for(var i = 0; i <= hacker1.length-1; i++){
+toUpperCaseString += hacker1[i] + " " ;
+}
+console.log(toUpperCaseString.toUpperCase());
+
+var reverseString = "";
+for(var i = hacker2.length-1; i >= 0; i--){
+reverseString += hacker2[i];
+}
+console.log(reverseString);
+
+var names = [hacker2, hacker1];
+var sortedName = names.sort();
+
+if (hacker1 === hacker2){
+  console.log("What?! You both got the same name?")
+} else if (sortedName[0] === hacker1){
+  console.log("The driver's name goes first")
+} else if (sortedName[0] === hacker2){
+  console.log("Yo, the navigator goes first definitely")
+}
 
 // Lorem ipsum generator
+
+
+var userString = prompt("Enter anything");
+
+var userString = userString.replace(' ', '');
```

Comment:
If you wanted to replace ALL the commas and spaces the easiest way would be with a regex (they can get SUPER confusing). 

```
var userString = userString.replace(/ /g, "");
var userString = userString.replace(/,/g, "");
```

the two / / means that what you are seeing is a regex. inside you put the code. In the first line, you can see you are replacing the space with nothing, and in the second example you can see the comma that again you are replacing with nothing. the g at the end stands for global (I think), which searches the entire string and replaces everything. 

If you wanted to, you could have done this in a for loop and checked each character to see if was a space or comma.